allennlp/data/vocabulary.py

```diff
@@ -150,13 +153,68 @@ def __init__(self,
                     if count >= min_count:
                         self.add_token_to_namespace(token, namespace)
 
+    def save_to_files(self, directory: str) -> None:
+        """
+        Persist this Vocabulary to files so it can be reloaded later.
+        Each namespace corresponds to one file.
+
+        Parameters
+        ----------
+        directory : ``str``
+            The directory where we save the serialized vocabulary.
+        """
+
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+
+        # warn if isn't empty
+        if os.listdir(directory):
+            logging.warning("vocabulary serialization directory % is not empty", directory)
+
+        for namespace, mapping in self._index_to_token.items():
+            # each namespace gets written to its own file, in index order
+            with gzip.open(os.path.join(directory, namespace), 'wt', encoding='utf-8') as token_file:
+                num_tokens = len(mapping)
+                for i in range(num_tokens):
+                    print(mapping[i], file=token_file)
+
+    @classmethod
+    def from_files(cls,
+                   directory: str,
```

Comment:
I think if we merge the logic for `set_from_file` and `from_files`, so that `from_files` just calls `set_from_file`, handling the case that you're just passed a single vocabulary file is trivial - you just use `set_from_file`.  We should probably merge the names a bit better, and make the documentation clear on this point.